### PR TITLE
[chore] Code linter findings for internal/aws

### PR DESCRIPTION
**Pull request by atoulme**
preallocate slices
do not assign a boolean if it is reassigned right after

### CODE REVIEW

File: internal/aws/k8s/k8sclient/clientset.go [hunk 1]
```diff
@@ -107,9 +107,9 @@ func InitSyncPollTimeout(pollTimeout time.Duration) Option {
 }
 
 func getStringifiedOptions(options ...Option) string {
-	var opts []string
-	for _, option := range options {
-		opts = append(opts, option.name)
+	opts := make([]string, len(options))
```
Comment (songy23) [inline internal/aws/k8s/k8sclient/clientset.go:110]:
```suggestion
	opts := make([]string, 0, len(options))
```

File: internal/aws/metrics/metric_calculator.go [hunk 2]
```diff
@@ -80,10 +80,12 @@ type Key struct {
 }
 
 func NewKey(metricMetadata interface{}, labels map[string]string) Key {
-	var kvs []attribute.KeyValue
+	kvs := make([]attribute.KeyValue, len(labels))
```
Comment (songy23) [inline internal/aws/metrics/metric_calculator.go:83]:
```suggestion
	kvs := make([]attribute.KeyValue, 0, len(labels))
```

**Review by songy23 — APPROVED** [review]

File: internal/aws/k8s/k8sclient/clientset.go (hunk 1 above)
Comment (atoulme) [reply internal/aws/k8s/k8sclient/clientset.go:110]: No, this is not warranted here.

File: internal/aws/metrics/metric_calculator.go (hunk 2 above)
Comment (atoulme) [reply internal/aws/metrics/metric_calculator.go:83]: OK here I agree this can use the append style.

Comment (atoulme) [reply internal/aws/metrics/metric_calculator.go:83]: addressed

**Review by dmitryax — APPROVED** [review]

**Pull request merged**